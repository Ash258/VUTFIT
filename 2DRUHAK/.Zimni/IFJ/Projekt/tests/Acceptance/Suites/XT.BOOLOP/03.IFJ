' Draw primary numbers from interval

declare function prvocislo(cislo as integer) as boolean

function prvocislo(a as integer) as boolean
    dim delitel as integer = a - 1

    do while delitel > 1

        if ((a \ delitel) * delitel) = a then
            return false
        else
        end if
        delitel = delitel - 1
    loop

    return true
end function

scope
    dim flag as boolean
    dim n1 as integer = 10
    dim n2 as integer = 20

    print !"Prime numbers between"; n1 ; !" and"; n2 ; !" are";

    dim i as integer = n1 + 1

    do while i < n2

        flag = prvocislo(i)

        if flag then
            print i;
        else
        end if

        i = i + 1
    loop
end scope
