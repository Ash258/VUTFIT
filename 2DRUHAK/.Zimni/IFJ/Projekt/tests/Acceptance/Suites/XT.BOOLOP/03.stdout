Prime numbers between 10 and 20 are 11 13 17 19